BUILD.md

```diff
@@ -52,6 +52,10 @@ There are numerous device backends, options, and architecture-specific optimizat
 ````
 which activates the OpenMP backend. All of the options controlling device backends, options, architectures, and third-party libraries (TPLs) are given below.
 
+Kokkos requires as a minimum C++17, however C++20 and C++23 are supported depending on the compiler.
```

Comment:
Sounds good but what does "C++20 and C++23 are supported depending on the compiler" actually mean?

Reply:
whether the compiler supports it?